app/workers/immediate_email_generation_worker.rb

```diff
@@ -60,22 +70,43 @@ def queue_for_priority(priority)
     end
   end
 
-  def subscription_contents
-    SubscriptionContent
-      .joins(:content_change, subscription: { subscriber: { subscriptions: :subscriber_list } })
-      .includes(:content_change, subscription: { subscriber: { subscriptions: :subscriber_list } })
-      .where.not(subscribers: { address: nil })
-      .where(email: nil)
+  def subscribers
+    SubscribersForImmediateEmailQuery.call
   end
 
-  def import_emails(subscription_contents)
-    subscription_content_changes = subscription_contents.map do |subscription_content|
-      {
-        subscription: subscription_content.subscription,
-        content_change: subscription_content.content_change,
-      }
+  def import_emails(subscribers)
+    email_params = subscribers.flat_map do |subscriber|
+      grouped_subscription_contents = subscriber
+        .unprocessed_subscription_contents
+        .group_by(&:content_change)
+
+      grouped_subscription_contents.map do |content_change_subscription|
+        {
+          address: subscriber.address,
+          content_change: content_change_subscription[0],
+          subscriptions: content_change_subscription[1].map(&:subscription)
+        }
+      end
     end
 
-    ImmediateEmailBuilder.call(subscription_content_changes)
+    ImmediateEmailBuilder.call(email_params)
+  end
+
+  def group_subscription_contents_by_content_change(subscribers)
+    #returns
+    #
+    # {
+    #   content_change => [subscription_content, subscription_content],
+    #   content_change => [subscription_content]
+    # }
+    #
+    # which maps to one key per email created
+    #
+    subscribers.inject({}) do |params, subscriber|
+      params.merge!(
+        subscriber.unprocessed_subscription_contents.group_by(&:content_change)
+      )
+      params
```

Comment:
I suspect you don't need this as the `merge!` will return the hash.

Reply:
Fixed in c7acb34